sdk/src/wallet/operations/transaction/high_level/delegation/create.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::{
+        address::{AccountAddress, Bech32Address},
+        context_input::{CommitmentContextInput, ContextInput},
+        output::{unlock_condition::AddressUnlockCondition, DelegationId, DelegationOutputBuilder, Output},
+    },
+    wallet::{
+        operations::transaction::TransactionOptions,
+        types::{OutputData, TransactionWithMetadata},
+        Wallet,
+    },
+};
+
+/// Params for `create_delegation_output()`
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateDelegationParams {
+    /// Bech32 encoded address which will control the delegation. Default will use the
+    /// ed25519 wallet address
```

Comment:
Maybe put a TODO if the default shouldn't be the account address (would depend on setting one account as preferred/default account)

Reply:
not sure what you mean

Reply:
I think we might want to have the address of an account output as remainder instead of the ed25519 address that controls the account

Reply:
do you think it should go like this then?
1. provided address
2. first wallet account address
3. wallet ed25519

Reply:
Yes, I just opened an issue for this, so we can resolve it here https://github.com/iotaledger/iota-sdk/issues/1888